stubs/cffi/_cffi_backend.pyi

```diff
@@ -0,0 +1,286 @@
+import sys
+import types
+from _typeshed import Incomplete, ReadableBuffer, WriteableBuffer
+from collections.abc import Callable, Hashable
+from typing import Any, ClassVar, Protocol, TypeVar, overload
+from typing_extensions import Literal, TypeAlias, final
+
+_T = TypeVar("_T")
+
+class _Allocator(Protocol):
+    def __call__(self, cdecl: str | CType, init: Any = ...) -> _CDataBase: ...
+
+FFI_CDECL: int
+FFI_DEFAULT_ABI: int
+RTLD_DEEPBIND: int
+RTLD_GLOBAL: int
+RTLD_LAZY: int
+RTLD_LOCAL: int
+RTLD_NODELETE: int
+RTLD_NOLOAD: int
+RTLD_NOW: int
+
+@final
+class CField:
+    bitshift: Incomplete
+    bitsize: Incomplete
+    flags: Incomplete
+    offset: Incomplete
+    type: Incomplete
+
+@final
+class CLibrary:
+    def close_lib(self, *args, **kwargs): ...
+    def load_function(self, *args, **kwargs): ...
+    def read_variable(self, *args, **kwargs): ...
+    def write_variable(self, *args, **kwargs): ...
+
+@final
+class CType:
+    abi: Incomplete
+    args: Incomplete
+    cname: Incomplete
+    elements: Incomplete
+    ellipsis: Incomplete
+    fields: Incomplete
+    item: Incomplete
+    kind: Incomplete
+    length: Incomplete
+    relements: Incomplete
+    result: Incomplete
+    def __dir__(self): ...
+
+@final
+class Lib:
+    def __dir__(self): ...
+
+@final
+class _CDataBase:
+    __name__: ClassVar[str]
+    def __add__(self, other): ...
+    def __bool__(self): ...
+    def __call__(self, *args, **kwargs): ...
+    def __complex__(self): ...
+    def __delitem__(self, other): ...
+    def __dir__(self): ...
+    def __enter__(self): ...
+    def __eq__(self, other): ...
+    def __exit__(self, type, value, traceback): ...
+    def __float__(self): ...
+    def __ge__(self, other): ...
+    def __getitem__(self, index): ...
+    def __gt__(self, other): ...
+    def __hash__(self): ...
+    def __int__(self): ...
+    def __iter__(self): ...
+    def __le__(self, other): ...
+    def __len__(self): ...
+    def __lt__(self, other): ...
+    def __ne__(self, other): ...
+    def __radd__(self, other): ...
+    def __rsub__(self, other): ...
+    def __setitem__(self, index, object): ...
+    def __sub__(self, other): ...
+
+@final
+class __CDataFromBuf(_CDataBase): ...  # type: ignore[misc]
```

Comment:
We _generally_ don't include stubs for private things unless there's a specific reason to make an exception. This is to lower the maintenance burden for us, and because a variety of things in our test suite assume that class starting with private names are things that shouldn't be tested, since they're usually "private to the stub" things like `TypedDict`s, `Protocol`s and type aliases.

I know very little about `cffi` -- is there a reason why it's important to include stubs for `__CDataFromBuf` (etc.)? (`_CData` is fine because we need it in the stub in order to type the `FFI` class correctly.)

Reply:
I think I can delete all of those except for _CDataBase.